doc/src/manual/arrays.md

```diff
@@ -872,10 +872,14 @@ slower than multiplication. While some arrays — like [`Array`](@ref) itself 
 are implemented using a linear chunk of memory and directly use a linear index
 in their implementations, other arrays — like [`Diagonal`](@ref) — need the
 full set of cartesian indices to do their lookup (see [`IndexStyle`](@ref) to
-introspect which is which). As such, when iterating over an entire array, it's
-much better to iterate over [`eachindex(A)`](@ref) instead of `1:length(A)`.
-Not only will the former be much faster in cases where `A` is `IndexCartesian`,
-but it will also support [OffsetArrays](https://github.com/JuliaArrays/OffsetArrays.jl), too.
+introspect which is which). 
+
+!!! warnings
+
+    When iterating over an entire array, it is
+    better to iterate over [`eachindex(A)`](@ref) instead of `1:length(A)`.
```

Comment:
Not new to this PR, but "iterating over an ... array" seems a bit confusing, since you can `iterate` arrays but not like this.

Maybe it should say "iterating over all indices of an array". Maybe it should mention that you can iterate the array itself, which is probably better when you don't actually need the index.